### PR TITLE
Intl.Locale.getWeekInfo() fixes

### LINKED ISSUE #33645: minimalDays could use more explanation
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

The description as is left me confused if I should care about this information or not. It would be useful to explain what it's for.

### What did you expect to see?

Something like:

> For calculating a date's week date (ex. The 20th week of the year) which start on `firstDay`. In some locales the first week date can be as little as one day long, in others it may require four days to be considered the first week, leaving the first few days as part of the last week of the previous year.

### Do you have any supporting links, references, or citations?

- https://en.wikipedia.org/wiki/ISO_week_date
- https://en.wikipedia.org/wiki/Leap_week_calendar

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md
* Last commit: https://github.com/mdn/content/commit/fb85334ffa4a2c88d209b1074909bee0e0abd57a
* Document last modified: 2023-09-25T07:33:31.000Z

</details>

### LINKED ISSUE #33719: Part of the example provided is factually incorrect
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo

### What specific section or headline is this issue about?

Example

### What information was incorrect, unhelpful, or incomplete?

The last example provided is factually incorrect. It currently states:

```
const msBN = new Intl.Locale("ms-BN");
console.log(msBN.getWeekInfo()); // { firstDay: 7, weekend: [5, 7], minimalDays: 1 }
// Brunei weekend is Friday and Sunday but not Saturday
```

Running this in a web browser that supports `getWeekInfo()`, such as Safari, produces:

```
{firstDay: 1, weekend: [6, 7], minimalDays: 1}
```

Thus, the value for property `firstDay` is `1` and not `7` and the array value for `weekend` property is `[6,7]` and not `[5,7]`

The Unicode CLDR (Common Locale Data Repository) also confirms this.

For the value of `firstDay` property, see the [XML file supplementalData.xml at line 4676](https://github.com/unicode-org/cldr/blob/2dd06669d833823e26872f249aa304bc9d9d2a90/common/supplemental/supplementalData.xml#L4676) or, alternatively, the [weekData.json at line 81](https://github.com/unicode-org/cldr-json/blob/0876ec40e13d54c0a6b6456392802d4de7e059cb/cldr-json/cldr-core/supplemental/weekData.json#L81)

For the value of `weekend` property, see the [XML file supplementalData.xml at line 4723](https://github.com/unicode-org/cldr/blob/2dd06669d833823e26872f249aa304bc9d9d2a90/common/supplemental/supplementalData.xml#L4723) which does not contain the territory `BN`, or alternatively, the [JSON file weekData.json at line 212](https://github.com/unicode-org/cldr-json/blob/0876ec40e13d54c0a6b6456392802d4de7e059cb/cldr-json/cldr-core/supplemental/weekData.json#L212)

### What did you expect to see?

If you really want to pick an outlier, then pick the locale `ar-AF` whose weekend starts on Thursday and ends on Friday:

```
new Intl.Locale('ar-AF').getWeekInfo();
// {firstDay: 6, weekend: [4, 5], minimalDays: 1}
```

Otherwise, it is best to remove this last example, since it is almost identical (except for the value `minDays` as the locale `en-GB` mentionned just above it.

### Do you have any supporting links, references, or citations?

The Unicode Common Locale Data Repository itself with the links provided.

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md
* Last commit: https://github.com/mdn/content/commit/fb85334ffa4a2c88d209b1074909bee0e0abd57a
* Document last modified: 2023-09-25T07:33:31.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
- Fix https://github.com/mdn/content/issues/33719
- Fix https://github.com/mdn/content/issues/33645

cc @jackdeguest @jamiebuilds in case you are interested in taking a look

### CODE REVIEW

**Comment by jackdeguest** [comment]
Thank you @Josh-Cena Lookds great !

File: files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md [hunk 1]
```diff
@@ -39,18 +39,20 @@ An object representing week information associated with the Locale data specifie
...
 
-const enGB = new Intl.Locale("en-GB");
+const enGB = new Intl.Locale("en-GB"); // English (United Kingdom)
 console.log(enGB.getWeekInfo()); // { firstDay: 1, weekend: [6, 7], minimalDays: 4 }
 
-const msBN = new Intl.Locale("ms-BN");
-console.log(msBN.getWeekInfo()); // { firstDay: 7, weekend: [5, 7], minimalDays: 1 }
-// Brunei weekend is Friday and Sunday but not Saturday
+const arAF = new Intl.Locale("ar-AF"); // Arabic (Afghanistan)
+console.log(arAF.getWeekInfo()); // { firstDay: 6, weekend: [4, 5], minimalDays: 1 }
```
Comment (jamiebuilds) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md:52]: I'm not sure why you'd omit an example where the weekend is not contiguous. That in combination with the note above seems to encourage people to ignore places where the weekend is split up. If you're going to use this API at all you might as well use it as intended

Comment (jamiebuilds) [reply files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md:52]: I see the other issue now, the underlying CLDR dataset is wrong. I think in this case there should be clarification and an issue filed in the cldr repo.



image

Comment (jackdeguest) [reply files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md:52]: > I see the other issue now, the underlying CLDR dataset is wrong. I think in this case there should be clarification and an issue filed in the cldr repo as the weekend in Brunei is on Friday and Sunday.

You can report an issue to the Unicode CLDR here: [https://cldr.unicode.org/](https://cldr.unicode.org/)

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/intl/locale/getweekinfo/index.md:52]: Yeah, thanks for the link. Until the bug is fixed (and browsers ship it) we can keep it in this form.

**Review by wbamberg — APPROVED** [review]
👍

**Pull request merged**